src/mono/dbi/cordb-process.cpp

```diff
@@ -0,0 +1,651 @@
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - EnableNGENPolicy - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT
+CordbProcess::Filter(const BYTE                             pRecord[],
+                     DWORD                                  countBytes,
+                     CorDebugRecordFormat                   format,
+                     DWORD                                  dwFlags,
+                     DWORD                                  dwThreadId,
+                     ICorDebugManagedCallback*              pCallback,
+                     /* [out][in] */ CORDB_CONTINUE_STATUS* pContinueStatus)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - Filter - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::ProcessStateChanged(CorDebugStateChange eChange)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - ProcessStateChanged - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::SetEnableCustomNotification(ICorDebugClass* pClass, BOOL fEnable)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - SetEnableCustomNotification - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::GetID(DWORD* pdwProcessId)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - GetID - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::GetHandle(HPROCESS* phProcessHandle)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - GetHandle - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::GetThread(DWORD dwThreadId, ICorDebugThread** ppThread)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - GetThread - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::EnumerateObjects(ICorDebugObjectEnum** ppObjects)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - EnumerateObjects - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::IsTransitionStub(CORDB_ADDRESS address, BOOL* pbTransitionStub)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - IsTransitionStub - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::IsOSSuspended(DWORD threadID, BOOL* pbSuspended)
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - IsOSSuspended - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::GetThreadContext(DWORD threadID, ULONG32 contextSize, BYTE context[])
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - GetThreadContext - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::SetThreadContext(DWORD threadID, ULONG32 contextSize, BYTE context[])
+{
+    LOG((LF_CORDB, LL_INFO100000, "CordbProcess - SetThreadContext - NOT IMPLEMENTED\n"));
+    return S_OK;
+}
+
+HRESULT CordbProcess::ReadMemory(CORDB_ADDRESS address, DWORD size, BYTE buffer[], SIZE_T* read)
+{
+    memcpy(buffer, (void*)address, size);
+    if (read != NULL)
+        *read = size;
+    LOG((LF_CORDB, LL_INFO1000000, "CordbProcess - ReadMemory - IMPLEMENTED\n"));
```

Comment:
This appears to be reading local process memory but the method is intended to read memory from the process being debugged.

Reply:
The process that we are debugging is remote, probably on Android or iOs. 
All the CORDB_ADDRESS that we are returning are array of bytes that came from the process that is being debugged and now is stored temporarily on mscordbi, that is why when it calls ReadMemory we are reading address from the local process memory.

Reply:
I suspect this is going to run into trouble as CORDB_ADDRESS was originally intended to be the address in the process being debugged. For example if a VS user opens the memory window and types in an arbitrary address I'd expect VS to do:
````C++
CORDB_ADDRESS debuggeeAddress = (CORDB_ADDRESS)userProvidedValueFromUI;
pProcess->ReadMemory(debuggeeAddress, ... );
````

Reply:
Thanks for the explanation, changed CORDB_ADDRESS to be an address from the debugged process. When ReadMemory is called I read on the debugged process.